docs/design-doc/Draft: Ambiguous KDoc references.md

```diff
@@ -0,0 +1,429 @@
+﻿#  Ambiguous KDoc references
+
+**Preamble:** The aim of the document is to address the current issues with ambiguous KDoc references that Dokka experiences during the migration to K2 (Analysis API). However, a chosen solution can be applied for future use cases as well.
+
+# Introduction
+
+There are two types of KDoc references to declaration:
+-   Fully qualified ones, for example `[com.example.classA]`
+-   Relative ones, for example `[memberProperty]`
+
+Also, KDoc allows to refer to:
+-   Parameters `[p]`.  They can be no only in `@param [p]` or `@param p`.
+-   A receiver via `[this]`
+
+Here is an example for understating:
+```kotlin
+package com.example
+/**  
+* [com.example.classA], [com.example.classA.member] - fully qualified references
+* [classA], [member], [extension] - relative reference
+*/
+class classA {
+	/**  
+	* [classA], [member], [extension] - relative reference
+	*/
+	val member = 0
+}
+  
+/**  
+ * [com.example.extension] - fully qualified reference 
+ * [classA], [extension] - relative references in the current scope
+ * [classA.extension], [com.example.classA.extension] - out of the scope of the current document
+ * [this] - receiver 
+ * [p] - parameter 
+ * 
+ * @param p is also a reference  
+ */
+ fun classA.extension(p: Int) = 0
+```  
+*Note: This document does not consider the case of extensions (references to extensions)  for the sake of simplicity.  It is a non-goal and can deserve another dedicated document.*
+
+
+# Problem
+
+For relative references, there some are cases when KDoc references are ambiguous that means there more than one possible candidate from the users point of view. These cases was discovered by the migration Dokka to K2 and behave differently in K1 and K2. ([original issue](https://github.com/Kotlin/dokka/issues/3451))
+
+### I Reference to itself
```

Comment:
is there a reason why `I` and `II` used in headers instead of `1` and `2`? just numbers are easier to read IMO